src/simtools/production_configuration/limits_calculation.py

```diff
@@ -181,22 +225,91 @@ def compute_viewcone(self, loss_fraction):
         )
-        bin_edges, bin_edges_unit = self.get_bin_edges_and_units(
-            self.angle_to_observing_position__triggered_showers_, axis="x"
+        return upper_bin_edge_value * u.deg
+
+    def plot_data(self):
+        """Plot the core distances and energies of triggered events."""
+        num_files = len(self.list_of_files)
+        showers_per_file = len(self.simulated) // num_files
+        shower_id_triggered_adjusted = self.adjust_shower_ids_across_files(
+            self.shower_id_triggered, num_files, showers_per_file
         )
-        if bin_edges_unit == "":
-            bin_edges_unit = "deg"
-        _, upper_bin_edge_value = self._compute_limits(
-            angle_to_observing_position__triggered_showers,
-            bin_edges,
-            loss_fraction,
-            axis=0,
-            limit_type="upper",
+
+        if self.telescope_list is not None:
+            mask = np.array(
+                [
+                    all(tel in event for tel in self.telescope_list)
+                    for event in self.trigger_telescope_list_list
+                ]
+            )
+            shower_id_triggered_adjusted = shower_id_triggered_adjusted[mask]
+
+        core_distances_all = np.sqrt(self.event_x_core**2 + self.event_y_core**2)
+        core_distances_triggered = core_distances_all[shower_id_triggered_adjusted]
+        triggered_energies = self.simulated[shower_id_triggered_adjusted]
+
+        core_bins = np.linspace(
+                core_distances_triggered.min(), core_distances_triggered.max(), 100
+            )
+        energy_bins = np.logspace(
+                np.log10(triggered_energies.min()), np.log10(triggered_energies.max()), 100
+            )
+        plt.figure(figsize=(8, 6))
+        plt.hist2d(
+            core_distances_triggered,
+            triggered_energies,
+            bins=[core_bins, energy_bins],
+            norm="log",
+            cmap="viridis",
         )
-        return upper_bin_edge_value * u.Unit(bin_edges_unit)
+
+        plt.colorbar(label="Event Count")
+        plt.xlabel("Core Distance [m]")
+        plt.ylabel("Energy [TeV]")
+        plt.yscale("log")
+        plt.title("2D Histogram of Triggered Core Distance vs Energy")
+        plt.show()
+
+    @staticmethod
+    def adjust_shower_ids_across_files(shower_id_triggered, num_files, showers_per_file):
+        """
+        Adjust shower_id_triggered by incrementing the IDs for each file so they don't repeat.
+
+        Parameters
+        ----------
+        shower_id_triggered : np.ndarray
+            Array of shower IDs for triggered events.
+        num_files : int
+            Number of files.
+        showers_per_file : int
+            Number of showers per file.
+
+        Returns
+        -------
+        np.ndarray
+            Adjusted shower IDs.
+        """
+        adjusted_ids = shower_id_triggered.copy()
+        for file_idx in range(1, num_files):
```

Comment:
Do I understand it correctly that this file modifies the array of all triggered events for each file? So it grows O(num_files * N), with many unneeded edits?

There is no unit test, but I think this would do the same with O(N) operations only:

```
    file_indices = shower_id_triggered // showers_per_file  # Compute file index for each shower ID
    return shower_id_triggered + file_indices * showers_per_file
 ```

Reply:
OK, but you agree that the initial implementation is with O(num_files * N) suboptimal and you modify the same values multiple times?